### ISSUE TITLE
No .json file generated when crawling directory.

### ISSUE THREAD

**Issue by myyellowshoe** [apollographql/persistgraphql#32]
I'm not getting a .json file pointing it to a directory, like so:
`$ persistgraphql client/src`

When I point it to a file it seams to work fine. Am I missing something here?

_Thanks for the all the amazing work contributed to the graphql community!_

**Comment by lemccarthy** [comment]
I'm having the same problem, I'm crawling specifically using `persistgraphql folder/ --js --extension=jsx`, and if there is a single file inside one of those directories (different levels of nesting seem to trigger/not trigger it) that does not have a gql tag inside of it, the command makes a mention of this error:

    (node:1383) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): 
    TypeError: Reduce of empty array with no initial value
    (node:1383) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In 
    the future, promise rejections that are not handled will terminate the Node.js process with a non-
    zero exit code.

Calling it on individual files and so on, including .jsx files works fine, *if* they contain a gql query. If not, I get the same error. I could probably cobble together some master map with disparate files, but I would really prefer to be able to simply run it over the whole project.

**Comment by myyellowshoe** [comment]
Yeah agreed, I've thought of doing a similar thing, but running it over a whole folder structure would be far better.

**Comment by myyellowshoe** [comment]
@Poincare Just a small bump for some thoughts on this when you get a quick min. Thanks so much! 👍

**Comment by rbpinheiro** [comment]
It's not generating for me, pointing to a folder or single file. My queries are on separate .gql files.
When I add --add-typename it doesn't even crawl the folder.
Also, the readme is saying to use --add_typename, but the cli corrects me tells me to use --add-typename.

**Comment by Poincare** [comment]
@myyellowshoe @lemccarthy Sorry guys - this summer has been pretty busy but I'm back now. To clarify, the issue is that if there is a single file that *doesn't* contain a gql tag, then it fails because of an empty array being mishandled?

**Comment by lemccarthy** [comment]
Yes, I have many directories which contain gql tags, but also files that
don't (dumb react view components), and the presence of even a single one
without the tag causes the whole process to fail. If I set up a test
directory with only tagged files, it succeeds.

**Comment by myyellowshoe** [comment]
@Poincare All good! Really appreciate the work on this. Haven't tested myself without files that don't have gql tags, only got it to work with a single file.

**Comment by sepehr500** [comment]
I am having this issue as well. Is anybody looking into a fix?

**Comment by sepehr500** [comment]
@Poincare  I have pushed up a fix for the issue. @lemccarthy @myyellowshoe , feel free to try out my fix. It is a small amount of code, and I tried it out around the office by having people modify their `node_module` folder.

**Comment by sepehr500** [comment]
We should close this issue if nobody is having problems anymore.

**Comment by mdeltito** [comment]
We ran into this issue as well (same thing, only happened on a single developer's machine) and found it also occurs when there is an _empty_ directory somewhere in the input path.
